array/array_test.go

```diff
@@ -121,8 +121,27 @@ func TestString(t *testing.T) {
 	}
 }
 
+type A struct {
+}
+
+type B struct {
+	A
+}
+
+type C interface {
+	C() string
+}
+
+func (*A) C() string {
+	return "A"
+}
+
+func (*B) C() string {
+	return "B"
+}
+
 func TestNewStringFromBinaryArray(t *testing.T) {
-	alloc := &fluxmemory.ResourceAllocator{}
+	alloc := &fluxmemory.GcAllocator{ResourceAllocator: &fluxmemory.ResourceAllocator{}}
```

Comment:
Can you revert all of the test changes? I don't think it adds to these tests. I'd be up for adding specific tests for `GcAllocator` but I don't want these tests to be changed.